Update payment-api-overview (#1541)
bigcommerce/dev-docs: docs/api-docs/payments/payments-api-overview.md

**Is store credit supported?**

> Yes, the Payments API supports store credit usage under a few conditions. The store credit payment method is only available for an order when transacting in the store's default currency and when the shopper has an available store credit balance. Store credit is **not** available for orders created by guest shoppers.

Reviewer: ```suggestion
Yes, the Payments API supports the store credit payment method under the following conditions: 
- The shopper is transacting in the store's default currency. 
- The shopper has a positive store credit balance. 

Store credit is _not_ available for orders created by guest shoppers.
```
This is just a suggestion. A list might be more concise, but the way you have it also works. Your call! 

Changed not from bold to italics. Google style guide recommends italics to draw attention to specific words.
